dashboard/app/models/ability.rb

```diff
@@ -184,6 +184,11 @@ def initialize(user)
     if user.admin?
       can :manage, :all
 
+      # We don't want admins to accidentally try to index _all_ existing
+      # sections so constrain their access.
+      cannot :index, Section
+      can :index, Section, user_id: user.id
```

Comment:
It'd be a bigger change, but it might be safest to remove `:index` permission for everything from admins and explicitly restore it in places we want it, with appropriate constraints.

```ruby
if user.admin?
  can :manage, :all

  # Explicitly enumerate indexable resources to avoid admins accidentally causing slow queries
  cannot :index, :all
  can :index, Section, user_id: user.id
  can :index, Script
  #...etc
```
If we would like to pursue this approach it should be a separate PR with some manual testing around the result.